libyul/AsmAnalysis.cpp

```diff
@@ -463,9 +464,8 @@ void AsmAnalyzer::checkAssignment(Identifier const& _variable, YulString _valueT
 
 	if (!found)
 	{
-		m_success = false;
 		// Only add message if the callback did not.
-		if (numErrorsBefore == m_errorReporter.errors().size())
+		if (watcher())
 			declarationError(_variable.location, "Variable not found or variable not lvalue.");
 	}
```

Comment:
```suggestion
if (!found && watcher())
	// Only add message if the callback did not.
	declarationError(_variable.location, "Variable not found or variable not lvalue.");
```

BTW, github suggestions work a bit weird here for some reason. The above also includes lines you removed...